pkg/template/template.go

```diff
@@ -32,42 +33,84 @@ type ReleaseSuite struct {
 	UnifiedChangelog string
 }
 
-// TemplatesExtension is the extension used for templates that we can use
-// to glob-load into the templating engine.
-const TemplatesExtension = ".md"
+// MarkdownPartialsExt is the extension used for markdown partials glob matcher
+const MarkdownPartialsExt = ".md"
+
+// HTMLPartialsExt is the extension used for HTML partials glob matcher
+const HTMLPartialsExt = ".htm"
+
+// HTMLTemplateExt is the extension used for templates that create HTML
+// output files
+const HTMLTemplateExt = ".htm"
 
 // Engine is a templating generation object that can use partials and helpers
 type Engine struct {
-	BaseDir string
-}
-
-// Define helper methods for templates
-var funcMap = stdlibTemplate.FuncMap{
-	"toLower": strings.ToLower,
+	BaseDir     string
+	PartialsDir string
 }
 
 // New returns a new templating.Engine based on the specified root
 // template directory.
 func New(baseDir string) *Engine {
 	return &Engine{
-		BaseDir: baseDir,
+		BaseDir:     baseDir,
+		PartialsDir: filepath.Join(baseDir, "partials"),
 	}
 }
 
-// WriteChangelog uses a combination of the template path and a data structure
-// to create an output file based on that template.
-func (engine *Engine) WriteChangelog(templateName string,
+func (engine *Engine) renderTemplate(
+	templateName string,
 	templateData interface{},
-	outputPath string) error {
-
+	outputFile *os.File,
+) error {
 	templatePath := filepath.Join(engine.BaseDir, templateName)
-	partialsGlobPath := filepath.Join(engine.BaseDir, "partials", "*"+TemplatesExtension)
 
 	// Sanity check
 	if _, err := os.Stat(templatePath); err != nil {
 		return fmt.Errorf("Could not read template '%s'", templatePath)
 	}
 
+	log.Printf("Generating '%s' file from template '%s'...", outputFile.Name(), templatePath)
+
+	// Sadly while `html/template` and `text/template` share the same API, they
+	// use incompatible signatures and parameters so depending on the extension,
+	// we have to duplicate our work.
+	if filepath.Ext(templatePath) == HTMLTemplateExt {
+		// HTML template generation
+		funcMap := htmlTemplate.FuncMap{
```

Comment:
Couldn't the funcMap still be a variable that's consumed by htmlTemplate.funcMap? While we still have to generate the funcmap seperately each time, it's better if both the htmlTemplate.FuncMap and textTemplate.FuncMap refer to the same source.

Reply:
hmm... let me try that - that would be great if it works and I thought about it for a sec but it seemed like the object that is being created was different. either way, I'll update you in a few mins if that works or not.

Reply:
@BradleyBoutcher your suggestion worked 👍 - PR updated!